src/Fixer/ClassNotation/SelfStaticAccessorFixer.php

```diff
@@ -0,0 +1,129 @@
+<?php
+
+/*
+ * This file is part of PHP CS Fixer.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *     Dariusz Rumiński <dariusz.ruminski@gmail.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace PhpCsFixer\Fixer\ClassNotation;
+
+use PhpCsFixer\AbstractFixer;
+use PhpCsFixer\FixerDefinition\CodeSample;
+use PhpCsFixer\FixerDefinition\FixerDefinition;
+use PhpCsFixer\FixerDefinition\VersionSpecification;
+use PhpCsFixer\FixerDefinition\VersionSpecificCodeSample;
+use PhpCsFixer\Tokenizer\Token;
+use PhpCsFixer\Tokenizer\Tokens;
+use PhpCsFixer\Tokenizer\TokensAnalyzer;
+
+final class SelfStaticAccessorFixer extends AbstractFixer
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function getDefinition()
+    {
+        return new FixerDefinition(
+            'Inside a final class or anonymous class `self::X` should be preferred to `static::X`.',
+            [
+                new CodeSample(
+                    '<?php
+final class Sample
+{
+    private static $A = 1;
+
+    public function getBar()
+    {
+        return static::class.static::test().static::$A;
+    }
+
+    private static function test()
+    {
+        return \'test\';
+    }
+}
+'
+                ),
+                new VersionSpecificCodeSample(
+                    '<?php
+$a = new class() {
+    public function getBar()
+    {
+        return static::class;
+    }
+};
+',
+                    new VersionSpecification(70000)
+                ),
+            ]
+        );
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function isCandidate(Tokens $tokens)
+    {
+        return $tokens->isAnyTokenKindsFound([T_CLASS, T_STATIC, T_DOUBLE_COLON]);
+    }
+
+    public function getPriority()
+    {
+        // must be run after FinalInternalClassFixer
+        return -10;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function applyFix(\SplFileInfo $file, Tokens $tokens)
+    {
+        $tokensAnalyzer = new TokensAnalyzer($tokens);
+
+        $classIndex = $tokens->getNextTokenOfKind(0, [[T_CLASS]]);
+
+        while (null !== $classIndex) {
+            if (
+                $tokens[$tokens->getPrevMeaningfulToken($classIndex)]->isGivenKind([T_FINAL])
+                || $tokensAnalyzer->isAnonymousClass($classIndex)
+            ) {
+                $this->fixClass($tokens, $classIndex);
```

Comment:
```php
final class Foo 
{
    function bar() 
    {
        $c = new class() {
            function baz() 
            {
                static::class;
            }
        };
    }
}
```

In this case, `fixClass` is called two times. But `static::` is already replaced in first call, for outer class.
So I think this should be optimized, to not call `fixClass` again for inner classes.